tools/downloader/src/open_model_zoo/model_tools/converter.py

```diff
@@ -104,11 +109,30 @@ def main():
 
     args = parser.parse_args()
 
+    try:
+        telemetry = tm.Telemetry('Model Converter', tid='UA-194864834-1')
+        telemetry.start_session('md')
+
+        if args.all:
+            telemetry.send_event('md', 'converter_selection_mode', 'all')
+        if args.list:
+            telemetry.send_event('md', 'converter_selection_mode', 'list')
+        if args.name:
+            telemetry.send_event('md', 'converter_selection_mode', 'name')
+    except NameError:
+        pass
+
     mo_path = args.mo
     if mo_path is None:
         try:
             mo_path = Path(os.environ['INTEL_OPENVINO_DIR']) / 'deployment_tools/model_optimizer/mo.py'
         except KeyError:
+            try:
+                telemetry.send_event('md', 'converter_error', str({'error': 'Unable to locate Model Optimizer.', 'model': 'none'}))
```

Comment:
I assume that the third argument here is intended for statistical analysis, in which case I have several questions/suggestions about it:

1. Why do you use a stringified Python literal as the value? Wouldn't a more commonly used data interchange format (like JSON) work better?
2. There doesn't seem to be much reason to include a human-readable error message here; a shorter error code (e.g. `unable-to-locate-mo`) would work better.
3. What's the purpose of `'model': 'none'` here? First, it is redundant, and second, "none" here is a string which would be a valid model name, so it's misleading too.

Reply:
1. According to current telemetry realization `event_label` should be a string https://github.com/openvinotoolkit/telemetry/blob/master/src/main.py#L49
2. Changed error codes
3. According to https://github.com/openvinotoolkit/telemetry#readme we need to send all keys if we use dict as a label

Reply:
> According to current telemetry realization event_label should be a string

Okay, but this doesn't answer the second half of my question.

I guess this would be a question for @AnastasiaKazantaeva - the README claims that the label could be a dictionary yet the library only accepts a string. How are we supposed to reconcile this? Maybe the library could be modified to accept a dictionary, so that we don't have to manually format the data?

> According to https://github.com/openvinotoolkit/telemetry#readme we need to send all keys if we use dict as a label

Fair enough, but I think we should at least use `None` rather than `'none'` so as not to create ambiguity.

@AnastasiaKazantaeva What's the rationale for always sending all keys? And doesn't this policy mean that no keys can be added to an event in the next release (since then the analytics sent by the previous release will lack that key)?

Reply:
@IRDonch you may send None instead of 'none'. Rational here is to simplify automation. Library accept only strings, because GA accept strings. But as we want to know more that only one single parameter, we added dict. This dict helps to link parameters. If we send it in diff events - we cannot align them.

Reply:
> Rational here is to simplify automation.

How does this simplify it?

> Library accept only strings, because GA accept strings. But as we want to know more that only one single parameter, we added dict.

But if you recommend dicts, wouldn't it make sense for the library itself to support dict inputs, so that all components submit data in a standardized format?